lua/nvim-treesitter-textobjects/shared.lua

```diff
@@ -220,47 +214,80 @@ function M.find_best_range(bufnr, capture_string, query_group, filter_predicate,
   return best
 end
 
+---@param range Range4
+---@param row integer
+---@param col integer
+---@return boolean
+local function is_in_range(range, row, col)
```

Comment:
`vim.treesitter._range.contains` should be a better choice because it could replace `contains` beneath this function. I kept them only because they are originally methods of `TSTextObject.Range`.

Reply:
Right, it's not related to any node, sorry about the noise. As far as I can see, `contains` was only used in one place (`shared.textobject_at_points`), so maybe we can simply do without that abstraction?

Reply:
Okay...I found it could not work, but only because of a typo not related to this, I've fixed it and force-pushed.

https://github.com/nvim-treesitter/nvim-treesitter-textobjects/compare/734f9acc449ea3819a80c1ca32140c9a05bab53b..af0aad9c5498a1a7cbf7b2b5644ab3fad9b5c44e

Reply:
Another reason  I did not inline them is that they are for `Range4`, but the variables we have is `Range6` in the scope these functions be called. Also another reason for using `contains`, it handles all subtypes of `Range`. I think we could do it after making the decision of whether to use functions from `vim.treesitter._range` or finally choose a `Range{n}` for all of our functions.

Reply:
How feasible is fixing an `n` for all function signatures?

Reply:
The information we get directly from the query is `Range6`. It contains a byte-level location for files and some functions need it, so if we chose one single type for all functions, it must be `Range6`. But some functions do not need it, doing so would make it more cumbersome.

We cannot convert `Rnage4` to `Range6` without knowing which source the range refers to, the bytes locations can not be calculated by only a rows and columns tuple.

Reply:
On the other hand, I'm not very familiar with the latest `vim.treesitter` API. I'm not sure if rewriting these functions to utilize the new API would change this situation. So, for now, I haven't done anything.

Reply:
Well, the question is whether using `Range6` consistently (and ignoring bytes in some places) is more or less cumbersome than having to explicitly convert them all over the place ;)